test/src/api/commands/element/testSetPasswordCheckReports.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const path = require('path');
+const {Key} = require('selenium-webdriver');
+const common = require('../../../../common.js');
+const {settings} = common;
+const NightwatchClient = common.require('index.js');
+const MockServer  = require('../../../../lib/mockserver.js');
+const CommandGlobals = require('../../../../lib/globals/commands.js');
+
+describe('setPassword check', function() {
+
+  before(function(done) {
+    this.server = MockServer.init();
+    this.server.on('listening', () => done());
+  });
+
+  after(function(done) {
+    this.server.close(function() {
+      done();
+    });
+  });
+
+  it('client.setPassword() value redacted in rawHttpOutput', async function() {
+
+    MockServer.addMock({
+      url: '/wd/hub/session/1352110219202/element/0/value',
+      method: 'POST',
+      postdata: {
+        text: Key.NULL + 'password',
+        value: [Key.NULL, 'p', 'a', 's', 's', 'w', 'o', 'r', 'd']
+      },
+      response: {
+        sessionId: '1352110219202',
+        status: 0
+      }
+    });
+
+    MockServer.addMock({
+      url: '/wd/hub/session/1352110219202/element/0/clear',
+      method: 'POST',
+      response: {
+        sessionId: '1352110219202',
+        status: 0
+      }
+    });
+
+    const testsPath = [
+      path.join(__dirname, '../../../../sampletests/checkValueRedacted/passwordValueRedacted.js')
+    ];
+
+    const globals = {
+      calls: 0,
+      waitForConditionTimeout: 10,
+      waitForConditionPollInterval: 10,
+      reporter(results) {
+        const rawHttpOutput = results.modules.passwordValueRedacted.rawHttpOutput;
+        const requests = rawHttpOutput.filter((req) => req[1].includes('/element/0/value') && req[1].includes('Request POST'));
+        let foundRedactedText = false;
+        let didNotFindNonRedactedText = true;
+        for (var request of requests) {
+          if (request[2].includes('redacted_text')) {
+            foundRedactedText = true;
+          }
+          if (request[2].includes('non_redacted')) {
+            didNotFindNonRedactedText = false;
+          }
```

Comment:
It's not very clear at the first glance what you're trying to do here, can you maybe add some comments here to briefly explain that?

Reply:
Also, instead of `redacted_text` and `non_redacted`, it should simply be `secretpassword` and `simpletext` (not adding any space or special character in between the words to make sure that these appear as it is in the report).

Reply:
added a comment [9ad5d68](https://github.com/nightwatchjs/nightwatch/pull/4066/commits/9ad5d68ae6a1aa07a9b487d389e95de41976f225)